luigi/contrib/jupyter_notebook.py

```diff
@@ -0,0 +1,310 @@
+        default=-1
+    )
+
+    json_action = luigi.Parameter(
+        default='delete'
+    )
+
+    pars = {}
+
+    def _form_input(self):
+
+        if isinstance(self.input(), dict):
+            out = {
+                tag: list(map(
+                    lambda x: x.path, _get_values(self.input().get(tag)))
+                ) for tag in self.input().keys()
+            }
+        else:
+            out = [
+                list(map(lambda x: x.path, _get_values(req)))
+                for req in self.input()
+            ]
+
+        return out
+
+    def _form_output(self):
+
+        if isinstance(self.output(), dict):
+            out = {
+                tag: req.path for tag, req in self.output().items()
+            }
+        else:
+            out = [req.path for req in self.output()]
+
+        return out
+
+    def run(self):
+
+        # check arguments
+        if not self.nb_path:
+            raise TypeError(
+                'nb_path cannot be None; '
+                'nb_path must be a valid path to a Jupyter notebook'
+            )
+
+        if not os.path.exists(self.nb_path):
+            raise IOError(
+                "I can't find the Jupyter notebook %s" % self.nb_path
+            )
+
+        if not self.kernel_name:
+            raise TypeError(
+                'kernel_name cannot be None; '
+                'kernel_name must be the name of a valid Jupyter kernel'
+            )
+
+        self.timeout = int(self.timeout)
+
+        if self.json_action not in ['keep', 'delete']:
+            raise ValueError(
+                "json_action must be one of 'keep' or 'delete'"
+            )
+
+        # get notebook name
+        notebook_name = _get_file_name_from_path(self.nb_path)
+
+        # get task id
+        task_id = self.task_id
+
+        # set requires pars
+        self.pars['input'] = self._form_input()
+
+        # set output pars
+        self.pars['output'] = self._form_output()
+
+        # write pars to temporary file
+        tmp_file_path = os.path.join(
+            os.path.dirname(self.nb_path), '%s_%s.ipynbpars'
+            % (notebook_name, task_id)
+        )
+
+        with open(tmp_file_path, 'w') as parameters:
+            json.dump(self.pars, parameters)
+
+        # set environment variable with tmp_file_path
+        os.environ['PARS'] = tmp_file_path
+
+        # run notebook
+        logger.info('=== Running notebook: %s ===' % notebook_name)
+
+        nb_version = int(nbformat.__version__.split('.')[0])
+
+        try:
+            with open(self.nb_path, 'r') as nb:
+                nb = nbformat.read(nb, as_version=nb_version)
+
+            ep = ExecutePreprocessor(
+                timeout=self.timeout,
+                kernel_name=self.kernel_name
+            )
+
+            ep.preprocess(nb, {
+                'metadata': {
+                    'path': os.path.dirname(self.nb_path)
+                }
+            })
+
+            logger.info('=== Done! ===')
```

Comment:
Just pointing out that this is only displayed when the notebook executes correctly. You could add a line like `logger.info('=== Done with errors! ===')` to the `except` block.

Reply:
Thanks for pointing this out. Added logging in the `except` block.